### ISSUE TITLE
CA1869 is flagging lazily-initialized JsonSerializerOptions construction

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#6957]
### Analyzer

**Diagnostic ID**: [CA1869](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1869): `Cache and reuse 'JsonSerializerOptions' instances`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0-preview.23468.1

### Describe the bug

The analyzer is flagging a JsonSerializerOptions construction being passed to Interlocked.CompareExchange.

### Steps To Reproduce

```C#
using System.Text.Json;

internal class Program
{
    private static JsonSerializerOptions? s_options;

    private static string Serialize<T>(T value)
    {
        JsonSerializerOptions options =
            s_options ??
            Interlocked.CompareExchange(ref s_options, new JsonSerializerOptions() { WriteIndented = true }, null) ??
            s_options;

        return JsonSerializer.Serialize(value, options);
    }

    private static void Main() => Console.WriteLine(Serialize(new { Name = "John", Age = 30 }));
}
```

### Expected behavior

No diagnostic.

### Actual behavior

<img width="980" alt="image" src="https://github.com/dotnet/roslyn-analyzers/assets/2642209/0f031f98-4ebe-43c8-8a75-fdac72982358">

**Comment by ericstj** [comment]
cc @Jozkee

**Comment by eiriktsarpalis** [comment]
I'm also seeing a related false-positive when lifting options instances out of loops:

![image](https://github.com/dotnet/roslyn-analyzers/assets/2813363/66f0677c-dd1c-42de-a6f4-4fcab063e78a)

**Comment by Jozkee** [comment]
@eiriktsarpalis I think you are implying that the `for` implies multiple JsonSerializer calls and hence the analyzer should not warn, it currently is not aware of such pattern but usage-in-loop detection can most likely be incorporated.
